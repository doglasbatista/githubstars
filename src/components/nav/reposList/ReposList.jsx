import React from 'react';
import PropTypes from 'prop-types';

import RepoItem from '../repoItem/RepoItem';

import { Container } from './styles';

const ReposList = ({ list }) => {
  return (
    <Container>
      {list.map(repo => (
        <RepoItem key={repo.node.id} repoData={repo.node} />
      ))}
    </Container>
  );
};

ReposList.propTypes = {
  list: PropTypes.arrayOf(
    PropTypes.shape({
      node: PropTypes.shape({
        resourcePath: PropTypes.string.isRequired,
        description: PropTypes.string.isRequired,
        stargazers: PropTypes.shape({
          totalCount: PropTypes.number.isRequired,
        }).isRequired,
      }).isRequired,
    })
  ).isRequired,
};

export default ReposList;
