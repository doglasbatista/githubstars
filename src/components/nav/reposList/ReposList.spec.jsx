import React from 'react';
import { render, fireEvent } from '@testing-library/react';

import ReposList from './ReposList';

describe('<ReposList />', () => {
  const repoList = [
    {
      node: {
        id: 'MDEwOlJlcG9zaXRvcnkyMjMyMjA0MzA=',
        shortDescriptionHTML:
          'Babel macro for counting number of lines or words in files at compile time',
        resourcePath: '/coffee-cup/count.macro',
        stargazers: {
          totalCount: 17,
        },
        viewerHasStarred: false,
      },
    },
  ];
  it('renders "Load More" button when hasNextPage is true', () => {
    const container = render(
      <ReposList
        addStar={() => {}}
        fetchMoreData={() => {}}
        hasNextPage
        list={repoList}
      />
    );

    expect(container).toMatchSnapshot();
  });

  it('doesn\'t render "Load More" button when hasNextPage is false', () => {
    const container = render(
      <ReposList
        addStar={() => {}}
        fetchMoreData={() => {}}
        hasNextPage={false}
        list={repoList}
      />
    );

    expect(container).toMatchSnapshot();
  });

  it('calls fetchMoreData when user click on "Load More" button', () => {
    const fetchMoreData = jest.fn();
    const { getByText } = render(
      <ReposList
        addStar={() => {}}
        fetchMoreData={fetchMoreData}
        hasNextPage
        list={repoList}
      />
    );

    fireEvent.click(getByText('Load More'));

    expect(fetchMoreData).toHaveBeenCalledTimes(1);
  });
});
