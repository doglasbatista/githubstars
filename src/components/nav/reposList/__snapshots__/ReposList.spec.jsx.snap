// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReposList /> doesn't render "Load More" button when hasNextPage is false 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-gzVnrw cYRKYB"
      >
        <div
          class="sc-bwzfXH kTTlQV"
        >
          <div
            class="sc-ifAKCX cFlEyZ"
          >
            <h3
              class="sc-htpNat iPlMzQ"
            >
              coffee-cup / count.macro
            </h3>
            <p
              class="sc-bxivhb cTGxyp"
            >
              Babel macro for counting number of lines or words in files at compile time
            </p>
            <span
              class="sc-EHOje cfSGJM"
            >
              <svg
                fill="none"
                height="1em"
                viewBox="0 0 18 18"
                width="1em"
              >
                <path
                  d="M9 1l2.472 5.008L17 6.816l-4 3.896.944 5.504L9 13.616l-4.944 2.6L5 10.712 1 6.816l5.528-.808L9 1z"
                  stroke="#5152B6"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
              </svg>
              17
            </span>
          </div>
          <div
            class="sc-bZQynM dsnLjH"
          >
            <button
              class="sc-bdVaJa jJwJGJ"
              type="button"
            >
              star
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-gzVnrw cYRKYB"
    >
      <div
        class="sc-bwzfXH kTTlQV"
      >
        <div
          class="sc-ifAKCX cFlEyZ"
        >
          <h3
            class="sc-htpNat iPlMzQ"
          >
            coffee-cup / count.macro
          </h3>
          <p
            class="sc-bxivhb cTGxyp"
          >
            Babel macro for counting number of lines or words in files at compile time
          </p>
          <span
            class="sc-EHOje cfSGJM"
          >
            <svg
              fill="none"
              height="1em"
              viewBox="0 0 18 18"
              width="1em"
            >
              <path
                d="M9 1l2.472 5.008L17 6.816l-4 3.896.944 5.504L9 13.616l-4.944 2.6L5 10.712 1 6.816l5.528-.808L9 1z"
                stroke="#5152B6"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
            17
          </span>
        </div>
        <div
          class="sc-bZQynM dsnLjH"
        >
          <button
            class="sc-bdVaJa jJwJGJ"
            type="button"
          >
            star
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<ReposList /> doesn't render "Load More" button when hasNextPage is true 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-gzVnrw cYRKYB"
      >
        <div
          class="sc-bwzfXH kTTlQV"
        >
          <div
            class="sc-ifAKCX cFlEyZ"
          >
            <h3
              class="sc-htpNat iPlMzQ"
            >
              coffee-cup / count.macro
            </h3>
            <p
              class="sc-bxivhb cTGxyp"
            >
              Babel macro for counting number of lines or words in files at compile time
            </p>
            <span
              class="sc-EHOje cfSGJM"
            >
              <svg
                fill="none"
                height="1em"
                viewBox="0 0 18 18"
                width="1em"
              >
                <path
                  d="M9 1l2.472 5.008L17 6.816l-4 3.896.944 5.504L9 13.616l-4.944 2.6L5 10.712 1 6.816l5.528-.808L9 1z"
                  stroke="#5152B6"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
              </svg>
              17
            </span>
          </div>
          <div
            class="sc-bZQynM dsnLjH"
          >
            <button
              class="sc-bdVaJa jJwJGJ"
              type="button"
            >
              star
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-gzVnrw cYRKYB"
    >
      <div
        class="sc-bwzfXH kTTlQV"
      >
        <div
          class="sc-ifAKCX cFlEyZ"
        >
          <h3
            class="sc-htpNat iPlMzQ"
          >
            coffee-cup / count.macro
          </h3>
          <p
            class="sc-bxivhb cTGxyp"
          >
            Babel macro for counting number of lines or words in files at compile time
          </p>
          <span
            class="sc-EHOje cfSGJM"
          >
            <svg
              fill="none"
              height="1em"
              viewBox="0 0 18 18"
              width="1em"
            >
              <path
                d="M9 1l2.472 5.008L17 6.816l-4 3.896.944 5.504L9 13.616l-4.944 2.6L5 10.712 1 6.816l5.528-.808L9 1z"
                stroke="#5152B6"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
            17
          </span>
        </div>
        <div
          class="sc-bZQynM dsnLjH"
        >
          <button
            class="sc-bdVaJa jJwJGJ"
            type="button"
          >
            star
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<ReposList /> renders "Load More" button when hasNextPage is true 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-gzVnrw cYRKYB"
      >
        <div
          class="sc-bwzfXH kTTlQV"
        >
          <div
            class="sc-ifAKCX cFlEyZ"
          >
            <h3
              class="sc-htpNat iPlMzQ"
            >
              coffee-cup / count.macro
            </h3>
            <p
              class="sc-bxivhb cTGxyp"
            >
              Babel macro for counting number of lines or words in files at compile time
            </p>
            <span
              class="sc-EHOje cfSGJM"
            >
              <svg
                fill="none"
                height="1em"
                viewBox="0 0 18 18"
                width="1em"
              >
                <path
                  d="M9 1l2.472 5.008L17 6.816l-4 3.896.944 5.504L9 13.616l-4.944 2.6L5 10.712 1 6.816l5.528-.808L9 1z"
                  stroke="#5152B6"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
              </svg>
              17
            </span>
          </div>
          <div
            class="sc-bZQynM dsnLjH"
          >
            <button
              class="sc-bdVaJa jJwJGJ"
              type="button"
            >
              star
            </button>
          </div>
        </div>
        <button
          class="sc-bdVaJa jiKOYj"
          type="button"
        >
          Load More
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-gzVnrw cYRKYB"
    >
      <div
        class="sc-bwzfXH kTTlQV"
      >
        <div
          class="sc-ifAKCX cFlEyZ"
        >
          <h3
            class="sc-htpNat iPlMzQ"
          >
            coffee-cup / count.macro
          </h3>
          <p
            class="sc-bxivhb cTGxyp"
          >
            Babel macro for counting number of lines or words in files at compile time
          </p>
          <span
            class="sc-EHOje cfSGJM"
          >
            <svg
              fill="none"
              height="1em"
              viewBox="0 0 18 18"
              width="1em"
            >
              <path
                d="M9 1l2.472 5.008L17 6.816l-4 3.896.944 5.504L9 13.616l-4.944 2.6L5 10.712 1 6.816l5.528-.808L9 1z"
                stroke="#5152B6"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
            17
          </span>
        </div>
        <div
          class="sc-bZQynM dsnLjH"
        >
          <button
            class="sc-bdVaJa jJwJGJ"
            type="button"
          >
            star
          </button>
        </div>
      </div>
      <button
        class="sc-bdVaJa jiKOYj"
        type="button"
      >
        Load More
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
